src/dom/uid.js

```diff
@@ -6,7 +6,7 @@ export default function(name) {
 
 function Id(id) {
   this.id = id;
-  this.href = window.location.href + "#" + id;
+  this.href = window.location.href.replace(window.location.hash, "") + "#" + id;
```

Comment:
How about instead: 

```js
this.href = new URL(`#${id}`, location) + "";
```

Reply:
Neat. I didn't know that was a thing. That's much better!